tools/c7n_azure/c7n_azure/resources/access_control.py

```diff
@@ -37,6 +37,33 @@
 
 @resources.register('roleassignment')
 class RoleAssignment(QueryResourceManager):
+    """Role assignments map role definitions to principals. The Azure
+    object only contains the unique ID of the principal, however we
+    attempt to augment the object with the prinicpal name, display name
+    and type from AAD.
+
+    Augmenting with data from AAD requires executing account to have
+    permissions to read from the Microsoft AAD Graph. For Service Principal
+    Authorization the Service Principal must have the permissions to
+    `read all users' full profiles`. Azure CLI authentication will
+    provide the necessary permissions to run the policy locally.
+
+    Model:
+    `RoleAssignment <https://docs.microsoft.com/en-us/python/api/azure-mgmt-authorization/
+    azure.mgmt.authorization.v2015_07_01.models.role_assignment?view=azure-python>`_
+
+    :example:
+
+    .. code-block:: yaml
+        policies:
+            - name: role-assignment-owner
+              resource: azure.roleassignment
+              filters:
```

Comment:
If we treat these example as a reference, do we really need filters here?

Reply:
i think its good for reference examples to have useful examples which typically includes a filter to make a real world use case.